ci-operator/step-registry/ibmcloud/provision/byok/ibmcloud-provision-byok-commands.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output   
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..." 
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+    ibmcloud plugin list
+}
+
+
+function create_resource_group() {
+    local rg="$1"
+    echo "create resource group ... ${rg}"
+    "${IBMCLOUD_CLI}" resource group-create ${rg} || return 1
+    "${IBMCLOUD_CLI}" target -g ${rg} || return 1
+}
+
+ibmcloud_login
+
+## Create the instances for BYOK
+echo "$(date -u --rfc-3339=seconds) - Creating the instance for BYOK..."
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+resource_group="${CLUSTER_NAME}-rg"
+
+create_resource_group ${resource_group}
+
+instance_name="${CLUSTER_NAME}-kp"
+
+ibmcloud resource service-instance-create ${instance_name} kms tiered-pricing ${region}
```

Comment:
Once created, need to write the info into `ibm_kp.json` to avoid any following command failure lead the created resource are not recorded, just similar to https://github.com/openshift/release/pull/46776/files#diff-cffe187ab93a8344026825911b3ff0131a9ed56511287a2c27cf19e725ceb5c8R89